### PR TITLE
Add misp add email command

### PR DESCRIPTION
**Pull request by YuvHayun**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
[CIAC-961](https://jira-hq.paloaltonetworks.local/okta_login.jsp?RelayState=%2Fbrowse%2FCIAC-961)

## Description
After updating the Docker image [here](https://github.com/demisto/dockerfiles/pull/8047), added the **!misp-add-email-object** command and relevant tests.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

### CODE REVIEW

File: Packs/MISP/.secrets-ignore [hunk 1]
```diff
@@ -1,3 +1,9 @@
...
 1.2.4.4
-1.5.52.1
\ No newline at end of file
+1.5.52.1
+209.85.128.198
+04.10.05.43
+CAPz_z5hBQLj98G50S_ST8UDRiW8b0Umtd1K-iVWPDTFKbzhN4g@mail.gmail.com
+2002:a25:aba6:0:b0:641:3b6f:41ac
+2002:a81:2592:0:b0:2d1:1210:275e
+2002:a5e:c913:0:0:0:0:0
```
Comment (JudahSchwartz) [inline Packs/MISP/.secrets-ignore:9]: where did these ips come from? Is it dummy data?

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 2]
```diff
@@ -2907,7 +3003,7 @@ script:
     - contextPath: MISP.Warninglist.Lists
       description: Name of warninglists where the value was found.
       type: string
-  dockerimage: demisto/pymisp2:1.0.0.30164
+  dockerimage: devdemisto/pymisp2:1.0.0.31710
```
Comment (JudahSchwartz) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:3006]: reminder to switch to prod before merge

File: Packs/MISP/Integrations/MISPV3/MISPV3_test.py [hunk 3]
```diff
@@ -45,6 +46,69 @@
                                     (None, []),  # no tag was detected, no event returns
                                     ("104", ['149'])]  # 1 event includes the detected event's tag
 
+EXPECTED_ADD_OBJECT_OUTPUT = {'Object': {'ID': '39958', 'Name': 'email', 'MetaCategory': 'network',
```
Comment (JudahSchwartz) [inline Packs/MISP/Integrations/MISPV3/MISPV3_test.py:49]: lets extract this to a json file under test_data

File: Packs/MISP/Integrations/MISPV3/MISPV3_test.py [hunk 4]
```diff
@@ -621,3 +685,120 @@ def test_warninglist_response(mocker):
...
+        demisto,
+        'internalHttpRequest',
+        return_value={
+            'body': json.dumps({'key': 'value'}),
+            'headers': json.dumps({'key': 'value'})
+        }
+    )
+
+    def json_func():
+        return {'Object': {'id': '39958', 'name': 'email', 'meta-category': 'network', 'description':
```
Comment (JudahSchwartz) [inline Packs/MISP/Integrations/MISPV3/MISPV3_test.py:703]: this too

File: Packs/MISP/Integrations/MISPV3/MISPV3_test.py [hunk 5]
```diff
@@ -621,3 +685,120 @@ def test_warninglist_response(mocker):
...
+    - Running add_email_object command.
+    Then:
+    - Ensure that the extraction of the information occured correctly.
+    """
+    from MISPV3 import add_email_object
+    import pymisp
+    event_id = 1231
+    demisto_args: dict = {'entry_id': "", 'event_id': event_id}
+    mocker.patch.object(demisto, "getFilePath", return_value={
+                        "path": "test_data/test email object.eml"
```
Comment (JudahSchwartz) [inline Packs/MISP/Integrations/MISPV3/MISPV3_test.py:772]: lets add a couple more test files. You mentioned for example the file the playbook downloaded worked on one version while a gmail eml did not. Lets add both, and also an outlook example file

File: Packs/MISP/ReleaseNotes/2_1_8.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#### Integrations
+##### MISP v3
+- Added the ***misp-add-email-object*** command.
+- Updated the Docker image to: *devdemisto/pymisp2:1.0.0.31710*.
```
Comment (JudahSchwartz) [inline Packs/MISP/ReleaseNotes/2_1_8.md:4]: reminder to switch tag

**Review by JudahSchwartz — COMMENTED** [review]
Nice! Its important, since the version of the library we're using is so flaky, to add a few more test cases

File: Packs/MISP/.secrets-ignore (hunk 1 above)
Comment (YuvHayun) [reply Packs/MISP/.secrets-ignore:9]: yes, from the "test email object.eml" file

File: Packs/MISP/Integrations/MISPV3/MISPV3_test.py [hunk 7]
```diff
@@ -621,3 +622,82 @@ def test_warninglist_response(mocker):
...
+    from MISPV3 import add_email_object
+    import pymisp
+    event_id = 1231
+    demisto_args: dict = {'entry_id': "", 'event_id': event_id}
+    mocker.patch.object(demisto, "getFilePath", return_value={
+                        "path": "test_data/test_add_email_object_case_1.eml"
+                        })
+    mocker.patch.object(pymisp.api.PyMISP, "_prepare_request", return_value=get_response(404, mocker, "response_mock_case_1"))
+    pymisp.ExpandedPyMISP.global_pythonify = True
+    try:
```
Comment (JudahSchwartz) [inline Packs/MISP/Integrations/MISPV3/MISPV3_test.py:698]: prefer pytest.raises

```
    with pytest.raises(DemistoException) as exception_info:
        add_email_object(demisto_args)
    assert expected_failure in str(exception_info.value)
```

**Review by JudahSchwartz — APPROVED** [review]
Amazing! Note my last 2 comments here

**Pull request merged**